cincinnati/src/plugins/internal/graph_builder/release_scrape_dockerv2/registry/mod.rs

```diff
@@ -315,31 +316,36 @@ pub async fn fetch_releases(
         Arc::new(FuturesMutex::new(Vec::with_capacity(estimated_releases)))
     };
 
+    // releases containing signatures that we want to ignore
+    let ignored_sig_releases = Arc::new(AtomicUsize::new(0));
+    // releases that had some issues and were not included in the graph
+    let ignored_releases = Arc::new(AtomicUsize::new(0));
     tags.try_for_each_concurrent(concurrency, |tag| {
         let registry_client = registry_client.clone();
         let cache = cache.clone();
         let releases = releases.clone();
+        let sig_releases = ignored_sig_releases.clone();
+        let skip_releases = ignored_releases.clone();
 
         async move {
             let (arch, manifestref, mut layers_digests) =
-                match get_manifest_layers(tag.to_owned(), &repo, &registry_client).await
-                {
-                    // todo: we're ignoring the images which we do not understand with this change. 
-                    // this change is required because we dont want cincinnati to error out on encountering
-                    // cosign signatures.
-                    // unintended consequence of this change can be that cincinnati continues with a log when it encounters 
-                    // incorrect image whereas it should error out. Cincinnati won't include this image in the update graph.
-                    // We should ideally try to ignore only the cosign .sig signatures. this change acts a 
-                    // stopgap till we teach cincinnati to deal with signatures
+                match get_manifest_layers(tag.to_owned(), &repo, &registry_client).await {
                     Ok(result) => result,
                     Err(e) => {
-                        warn!(
-                            "error fetching manifest and manifestref for {}:{}: {}, ignoring this image",
-                            &repo,
-                            &tag,
-                            e
+                        if tag.contains(".sig") {
```

Comment:
[Tag-based discovery][1] is [one option][2] for finding Sigstore signatures.  In the future, we might move to [listing referrers][3] (containers/image#2030).  But if we do, the failure modes for this line:

* Misclassifying a non-sig as a Sigstore signature because it happens to use this tag structure, or
* Misclassifying a Sigstore signature as a non-sig, non-release ignored image,

both seem low, so :shrug:, I'm ok with this heuristic.

[1]: https://github.com/sigstore/cosign/blob/v2.4.0/specs/SIGNATURE_SPEC.md#tag-based-discovery
[2]: https://github.com/sigstore/cosign/blob/v2.4.0/specs/SIGNATURE_SPEC.md#discovery
[3]: https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md#listing-referrers

Reply:
for misclassifying non-sig to sig will always be a risk if we do string comparison. but imo should be rare. 
if signature gets classifies as non-sig, the logs should bring that into notice. Not a lot worried about the mismatch.
We can also change this logic when we get listing referrers in dkregistry and pull it downstream to cincinnati.